src/frontend/posts_page/components/PostBlock.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+
+interface PostBlockProps {
+  post: Post
+}
+
+interface Post {
+  name: string,
+  time: string,
+  detail: string
+}
+
+interface PostBlockState {
+  like: number,
+  liked: boolean
+}
+
+
+class PostBlock extends React.Component<PostBlockProps, PostBlockState> {
+  constructor(props: PostBlockProps) {
+    super(props);
+    this.state = {
+      like: 0,
+      liked: false
+    };
+  }
+
+  markLike = () => {
+    if(this.state.liked) {
+      this.setState({like: this.state.like - 1});
```

Comment:
I think Danya said to use a function (its in her slide) like the one below to get the previous state instead of doing `this.state.something`. The suggestion below also applies to all your `setState()` functions in your changed files that need to use the previous state. And I think if you don't do `{...state, like: state.like - 1}`, you might lose the state for `liked` because you are changing your state from `{like: a, liked: b}` to just `{like: a}` (<- not too sure, you can check)

```suggestion
      this.setState(state: PostBlockState => ({...state, like: state.like - 1}));
```

Reply:
My implementation here works for sure, and I believe the sample button she used during workshop used that form of setState as well.